### ISSUE TITLE
Clarify what we mean by negative tests

### ISSUE THREAD

**Issue by moekraft** [w3c/wcag-act#7]
Generally as test writers we perform negative tests:

- A car has doors. 
- Test: Check if car does not have doors.

Confusion expressed in [WCAG WG survey](https://www.w3.org/2002/09/wbs/35422/ACT_DPUB_Review_Nov_2016/results).

1. The document says (section 3.8 2.8) that all tests are negative tests, but that the testing can be automated/manual/combined - wouldn't it sometimes be easier to have a positive test (not always of course). Perhaps this is too limiting?
2. The document says (section 3.8 2.8) that the rule set will test for failures. Will these rules be based solely on our documented failure techniques? If so the rule set may be limited. Some people have given up trying to introduce failures. 

Need clarification with example in specification.

**Comment by moekraft** [comment]
Alistair will work on writing a clarification.

**Comment by moekraft** [comment]
Additional feedback from Bruce Bailey - I think the main idea behind “The ACT Framework results in negative feature tests” is okay.
This would be compatible with the principle that “not applicable” is a pass, and that SC are written as affirmative statements.

**Comment by moekraft** [comment]
Both positive and negative tests should be possibilities. - Jonathan Avila.

**Comment by moekraft** [comment]
From Alistair 11/16/16
Dear All, 


I have an action item to clarify the meaning of “negative” in the following section:
 

3.8. 2.8 Rules Test for Failures

 
The ACT Framework results in negative feature tests, meaning that ACT Rules test for violations instead of compliance. ACT Rules should map to [WCAG 2.0 Failure Techniques](https://www.w3.org/TR/WCAG20-TECHS/failures.html) where possible to avoid duplication of work. In some cases absence of violations may be proof of compliance, if rules are available to test all possible violations.

 

I would suggest the following re-write:
 

3.8. 2.8 Rules provide clear reasons for non-compliance

 
The ACT Framework will focus on defining rules that enable clear reasons for non-compliance to be given to the user e.g. “displayed content in a page flashes more than three times per second”.  Where possible, ACT Rules should map to [WCAG 2.0 Failure Techniques](https://www.w3.org/TR/WCAG20-TECHS/failures.html).


Interested to hear thoughts / comments.


Very best regards

 
Alistair

**Comment by moekraft** [comment]
Please note the relationship between this issue an issue #15, Mapping rules to WCAG 2.0 Failure Techniques is problematic.

**Comment by moekraft** [comment]
We will remove the restriction that test must be negative tests. See meeting minutes 4/10/2017.

**Comment by moekraft** [comment]
"It is important to know the difference though" between positive and negative tests. - Romain.

**Comment by moekraft** [comment]
***update*** After further discussion we have determined that we need to revisit this. Shadi will do more investigation on the difference between negative and positive tests.

**Comment by WilcoFiers** [comment]
*Negative tests* is no longer used as a term in the latest draft of ACT Rules format.
